src/test/java/com/example/onlinebookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,272 @@
+    @SneakyThrows
+    static void addUsers(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/user/add-users.sql")
+            );
+        }
+    }
+
+    @SneakyThrows
+    static void addCarts(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/cart/add-carts.sql")
+            );
+        }
+    }
+
+    @SneakyThrows
+    static void addBooks(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/book/add-five-books.sql")
+            );
+        }
+    }
+
+    @SneakyThrows
+    static void addItems(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/cartitem/add-cart-items.sql")
+            );
+        }
+    }
+
+    @SneakyThrows
+    static void deleteBooks(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/book/delete-books.sql")
+            );
+        }
+    }
+
+    @SneakyThrows
+    static void deleteCartItems(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/cartitem/delete-cart-items.sql")
+            );
+        }
+    }
+
+    @SneakyThrows
+    static void deleteUsers(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/user/delete-users.sql")
+            );
+        }
+    }
+
+    @SneakyThrows
+    static void deleteCarts(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/cart/delete-carts.sql")
+            );
+        }
+    }
```

Comment:
160 lines of code before I get to 1st test method.
Please move service methods to the bottom and make them private unless reused in other test classes

recheck all classes for same issue